tests/logitech_receiver/test_base.py

```diff
@@ -111,3 +118,35 @@ def test_get_next_sw_id():
 
     assert res1 == 2
     assert res2 == 3
+
+
+@pytest.mark.parametrize(
+    "simulated_error, expected_result",
+    [
+        (ERROR.invalid_SubID__command, 1.0),
+        (ERROR.resource_error, None),
+        (ERROR.connection_request_failed, None),
+        (ERROR.unknown_device, exceptions.NoSuchDevice),
+    ],
+)
+def test_ping(simulated_error, expected_result):
```

Comment:
Add this to skip test on macOS or patch the hidapi to make it work on macOS
`
import pytest
import sys

@pytest.mark.skipif(sys.platform == "darwin", reason="Test only runs on Linux")
`